### ISSUE TITLE
[java] EmptyControlStatement: False positive on concise try-with-resources

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6628]
**Affects PMD Version:** 7.23.0

**Rule:** EmptyControlStatement — https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#emptycontrolstatement

**Description:**

PMD reports `EmptyControlStatement` on the concise try-with-resources form introduced in Java 9 (`try (existingVar) { }`). However, this construct is not an empty control statement, since the resource is still properly closed by the runtime. The rule incorrectly treats this form as a normal empty `try` block, leading to a false positive,

**Code Sample demonstrating the issue:**

```java
import java.io.ByteArrayInputStream;
import java.io.InputStream;

public class EmptyTryResource {

    // flagged by EmptyControlStatement (false-positive)
    public void closeConcise() throws Exception {
        InputStream in = new ByteArrayInputStream(new byte[]{1, 2, 3});
        try (in) {                       // line 10: "Empty try-with-resources statement. Should be written in.close()"
        }
    }

    public void closeMultipleConcise() throws Exception {
        InputStream in1 = new ByteArrayInputStream(new byte[]{7, 8});
        InputStream in2 = new ByteArrayInputStream(new byte[]{9, 10});
        try (in1; in2) {                 // also flagged
        }
    }

}
```

**Actual outcome:** PMD reports a  `EmptyControlStatement` violation at line 10, but that's wrong. That's a false positive.

**Running PMD through:** CLI

**Comment by zbynek** [comment]
PMD does not treat them as empty try block. The message does not suggest removing them, but replacing them with `in.close()`. 
The rationale for flagging these blocks is explained in https://github.com/pmd/pmd/issues/4930 -- I don't think this is a false positive.

**Comment by Emilyaxe** [comment]
> [#4930](https://github.com/pmd/pmd/issues/4930)

Thank you for the kind explanation.

**Comment by adangel** [comment]
Yes, the rule works correctly and this is not a false positive.

Should we improve the rule documentation to explicitly mention this case (either in the rule description or give an example or both)?

**Comment by adangel** [comment]
I'm closing this meanwhile as not planned
